### DISCUSSION TITLE
Worker doesn't released until executing sleep(1) in the Main Thread

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#21873] [Q&A]
emcc 3.1.55

Worker is stuck on reading a file :
     `read -> _fd_read -> proxyToMainThread -> emscripten_futex_wait`

it released after a few (10) seconds.
but if we execute a` sleep(1)` in the main thread loop, it release immediately.

The file is not accessed by the main thread, how it can be explained ?

**Comment by aharondavid** [comment]
I figure it out:
the main thread,  is fire a lot of `proxyToMainThread  `calls for GL processing.

I wander why the `_fd_read  `need to be executed in the Main Thread while the file is placed entirely in the memory.

**Reply by sbc100** [reply]
I believe the new wasmfs filesystem (`-sWASMFS`) keep the memfs in shared memory and therefore doesn't need proxying to read/writing to memfs files.
